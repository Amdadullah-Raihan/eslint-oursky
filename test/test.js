const path = require("path");

const { ESLint } = require("eslint");

function runTest(filename) {
  test(filename, async () => {
    const cli = new ESLint();
    const lintResults = await cli.lintFiles([path.join("src", filename)]);
    const formatter = await cli.loadFormatter("unix");
    const actual = await formatter.format(lintResults);
    expect(actual).toMatchSnapshot();
  });
}

runTest("simple.ts");
runTest("react.tsx");
