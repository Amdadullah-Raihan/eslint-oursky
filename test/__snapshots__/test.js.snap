// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react.tsx 1`] = `
"src/react.tsx:3:26: Use VFC and VoidFunctionComponent instead of FC and FunctionComponent. [Error/@oursky/react-vfc]
src/react.tsx:7:26: Use VFC and VoidFunctionComponent instead of FC and FunctionComponent. [Error/@oursky/react-vfc]
src/react.tsx:11:26: Use VFC and VoidFunctionComponent instead of FC and FunctionComponent. [Error/@oursky/react-vfc]
src/react.tsx:15:26: Use VFC and VoidFunctionComponent instead of FC and FunctionComponent. [Error/@oursky/react-vfc]

4 problems"
`;

exports[`simple.ts 1`] = `
"src/simple.ts:1:1: Unexpected var, use let or const instead. [Error/no-var]
src/simple.ts:1:8: Expected a \`const\` assertion instead of a literal type annotation. [Error/@typescript-eslint/prefer-as-const]
src/simple.ts:5:8: Missing return type on function. [Error/@typescript-eslint/explicit-module-boundary-types]
src/simple.ts:6:11: Switch is not exhaustive. Cases not matched: \\"b\\" [Error/@typescript-eslint/switch-exhaustiveness-check]
src/simple.ts:12:1: Unexpected var, use let or const instead. [Error/no-var]
src/simple.ts:12:5: 'b' is assigned a value but never used. Allowed unused vars must match /^_/u. [Error/@typescript-eslint/no-unused-vars]
src/simple.ts:12:11: Unexpected number value in conditional. An explicit zero/NaN check is required. [Error/@typescript-eslint/strict-boolean-expressions]

7 problems"
`;
